src/discord/discord.service.ts

```diff
@@ -1,26 +1,73 @@
-import { Injectable } from '@nestjs/common';
+import { Injectable, NotFoundException } from '@nestjs/common';
 import { CreateDiscordDto } from './dto/create-discord.dto';
+import { ReadDiscordDto } from './dto/read-discord.dto';
 import { UpdateDiscordDto } from './dto/update-discord.dto';
 
 @Injectable()
 export class DiscordService {
+  private readonly discord: ReadDiscordDto[] = [];
   create(createDiscordDto: CreateDiscordDto) {
-    return 'This action adds a new discord';
+    const discordUser = {
+      id: Date.now(),
+      username: createDiscordDto.username,
+      bio: { ...createDiscordDto.bio },
+      createdOn: new Date(Date.now()),
+      updatedOn: new Date(Date.now()),
+    };
+    this.discord.push(discordUser);
+    return 'User added successfully!';
   }
 
   findAll() {
-    return `This action returns all discord`;
+    return [...this.discord];
   }
 
   findOne(id: number) {
-    return `This action returns a #${id} discord`;
+    const discordUser = this.findDiscord(id);
+
+    return { ...discordUser };
   }
 
   update(id: number, updateDiscordDto: UpdateDiscordDto) {
-    return `This action updates a #${id} discord`;
+    const {
+      username,
+      bio: { description, twitter, linkedin, github },
+    } = updateDiscordDto;
+
+    const discordUser = this.findDiscord(id);
+    const updatedDiscord = { ...discordUser };
+    if (username) {
+      updatedDiscord.username = username;
+    }
+    if (description) {
+      updatedDiscord.bio.description = description;
+    }
+    if (twitter) {
+      updatedDiscord.bio.twitter = twitter;
+    }
+    if (linkedin) {
+      updatedDiscord.bio.linkedin = linkedin;
+    }
+    if (github) {
+      updatedDiscord.bio.github = github;
+    }
+
+    return `User updated successfully!`;
```

Comment:
The return string on line `61` uses single quotes, it is better to be consistent with that if there are no variables in the string here